linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2115,6 +2116,64 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
             defaultApply(ClassType(ClassClass))
         }
 
+      case ClassGetName =>
+        newReceiver match {
+          case BlockOrAlone(stats, ClassOf(typeRef)) =>
+            def mappedClassName(className: ClassName): String = {
+              RuntimeClassNameMapperImpl.map(
+                  config.coreSpec.semantics.runtimeClassNameMapper,
+                  className.nameString)
+            }
+
+            val nameString = typeRef match {
+              case PrimRef(tpe) =>
+                tpe match {
+                  case NoType      => "void"
+                  case BooleanType => "boolean"
+                  case CharType    => "char"
+                  case ByteType    => "byte"
+                  case ShortType   => "short"
+                  case IntType     => "int"
+                  case LongType    => "long"
+                  case FloatType   => "float"
+                  case DoubleType  => "double"
+                  case NothingType => "nothing" // invalid
+                  case NullType    => "null"    // invalid
+                }
+
+              case ClassRef(className) =>
+                mappedClassName(className)
+
+              case ArrayTypeRef(PrimRef(tpe), dimensions) =>
+                val charCode = tpe match {
+                  case NoType      => "V"
+                  case BooleanType => "Z"
+                  case CharType    => "C"
+                  case ByteType    => "B"
+                  case ShortType   => "S"
+                  case IntType     => "I"
+                  case LongType    => "J"
+                  case FloatType   => "F"
+                  case DoubleType  => "D"
+                  case NothingType => "E" // invalid
+                  case NullType    => "N" // invalid
+                }
```

Comment:
Could we re-use `charCodeOfPrimRef` in the `CoreJSLib` here? We might even move it to `ir`....

If yes, maybe the info in `definePrimitiveTypeDatas` could also be consolidated.

OTOH, it's probably unavoidable that the optimizer has some duplicate code :-/

Reply:
I've added a `val charCode` in `PrimRef`, and used the existing `displayName` for the other match above.